### PR TITLE
feat(argo-cd): add extraContainers to pods besides server

### PR DESCRIPTION
**Pull request by mikutas**
Signed-off-by: mikutas <23391543+mikutas@users.noreply.github.com>

Similar to #465 

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Comment by mbevc1** [comment]
Wouldn't that need to be added outside of current container spec? :thinking:

File: charts/argo-cd/templates/argocd-application-controller/deployment.yaml [hunk 1]
```diff
@@ -106,6 +106,9 @@ spec:
         {{- end }}
         resources:
 {{- toYaml .Values.controller.resources | nindent 10 }}
+    {{- if .Values.controller.extraContainers }}
+{{- toYaml .Values.controller.extraContainers | nindent 6 }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/argocd-application-controller/deployment.yaml:110]: Please DRY and use:
```suggestion
    {{- with .Values.controller.extraContainers }}
      {{- toYaml . | nindent 6 }}
```

File: charts/argo-cd/templates/argocd-repo-server/deployment.yaml [hunk 2]
```diff
@@ -114,6 +114,9 @@ spec:
           failureThreshold: {{ .Values.repoServer.readinessProbe.failureThreshold }}
         resources:
 {{- toYaml .Values.repoServer.resources | nindent 10 }}
+    {{- if .Values.controller.extraContainers }}
+{{- toYaml .Values.controller.extraContainers | nindent 6 }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/argocd-repo-server/deployment.yaml:118]:
```suggestion
    {{- with .Values.controller.extraContainers }}
      {{- toYaml . | nindent 6 }}
```

File: charts/argo-cd/templates/dex/deployment.yaml [hunk 3]
```diff
@@ -100,6 +100,9 @@ spec:
         {{- end }}
         resources:
 {{- toYaml .Values.dex.resources | nindent 10 }}
+    {{- if .Values.dex.extraContainers }}
+{{- toYaml .Values.dex.extraContainers | nindent 6 }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/dex/deployment.yaml:104]:
```suggestion
    {{- with .Values.dex.extraContainers }}
      {{- toYaml . | nindent 6 }}
```

File: charts/argo-cd/templates/redis/deployment.yaml [hunk 4]
```diff
@@ -64,6 +64,9 @@ spec:
 {{- end }}
         resources:
 {{- toYaml .Values.redis.resources | nindent 10 }}
+    {{- if .Values.redis.extraContainers }}
+{{- toYaml .Values.redis.extraContainers | nindent 6 }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/redis/deployment.yaml:68]:
```suggestion
    {{- with .Values.redis.extraContainers }}
      {{- toYaml . | nindent 6 }}
```

**Review by mkilchhofer — CHANGES_REQUESTED** [review]

**Review by mbevc1 — APPROVED** [review]

**Review by mkilchhofer — APPROVED** [review]
lgtm

**Pull request merged**